diesel/src/insertable.rs

```diff
@@ -33,18 +45,71 @@ pub enum ColumnInsertValue<Col, Expr> {
     Default(Col),
 }
 
+impl<Col, Expr, DB> InsertValues<Col::Table, DB> for ColumnInsertValue<Col, Expr>
+where
+    DB: Backend + SupportsDefaultKeyword,
+    Col: Column,
+    Expr: Expression<SqlType = Col::SqlType> + QueryFragment<DB>,
+{
+    fn column_names(&self, out: &mut DB::QueryBuilder) -> QueryResult<()> {
+        out.push_identifier(Col::NAME)?;
+        Ok(())
+    }
+
+    fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
+        if let ColumnInsertValue::Expression(_, ref value) = *self {
+            value.walk_ast(out.reborrow())?;
+        } else {
+            out.push_sql("DEFAULT");
+        }
+        Ok(())
+    }
+
+    fn is_noop(&self) -> bool {
+        false
+    }
+}
+
+#[cfg(feature = "sqlite")]
+impl<Col, Expr> InsertValues<Col::Table, Sqlite> for ColumnInsertValue<Col, Expr>
+where
+    Col: Column,
+    Expr: Expression<SqlType = Col::SqlType> + QueryFragment<Sqlite>,
+{
+    fn column_names(&self, out: &mut <Sqlite as Backend>::QueryBuilder) -> QueryResult<()> {
+        if let ColumnInsertValue::Expression(..) = *self {
+            out.push_identifier(Col::NAME)?;
+        }
+        Ok(())
+    }
+
+    fn walk_ast(&self, mut out: AstPass<Sqlite>) -> QueryResult<()> {
+        if let ColumnInsertValue::Expression(_, ref value) = *self {
+            value.walk_ast(out.reborrow())?;
+        }
+        Ok(())
+    }
+
+    fn is_noop(&self) -> bool {
+        if let ColumnInsertValue::Expression(..) = *self {
+            false
+        } else {
+            true
+        }
```

Comment:
I'd probably flip this to `if let ColumnInsertValue ::Default(..) = *self { true } else { false }` (and maybe even add a method to the enum if we ever need that for another thing).

Reply:
:+1: will follow up with that once I'm done moving this code around